Source/ProgressionSystemRuntime/Private/Widgets/PSMenuWidget.cpp

```diff
@@ -11,11 +11,62 @@
 #include "PoolManagerSubsystem.h"
 #include "PoolManagerTypes.h"
 #include "Components/StaticMeshComponent.h"
+#include "GameFramework/MyGameStateBase.h"
+#include "GameFramework/MyPlayerState.h"
+#include "Subsystems/GlobalEventsSubsystem.h"
 #include "UtilityLibraries/MyBlueprintFunctionLibrary.h"
 
 
 #include UE_INLINE_GENERATED_CPP_BY_NAME(PSMenuWidget)
 
+// Called after the underlying slate widget is constructed.
+void UPSMenuWidget::NativeConstruct()
+{
+	Super::NativeConstruct();
+
+	// Hide this widget by default
+	SetVisibility(ESlateVisibility::Collapsed);
+
+	// Listen to handle input for each game state
+	BIND_ON_GAME_STATE_CHANGED(this, ThisClass::OnGameStateChanged);
+
+	// Binds the local player state ready event to the handler
+	BIND_ON_LOCAL_PLAYER_STATE_READY(this, ThisClass::OnLocalPlayerStateReady);
+}
+
+// Called when the current game state was changed
+void UPSMenuWidget::OnGameStateChanged_Implementation(ECurrentGameState CurrentGameState)
+{
+	switch (CurrentGameState)
+	{
+	case ECurrentGameState::GameStarting:
+		SetVisibility(ESlateVisibility::Collapsed);
+	default: return;
+	}
+}
+
+// Subscribes to the end game state change notification on the player state
+void UPSMenuWidget::OnLocalPlayerStateReady_Implementation(AMyPlayerState* PlayerState, int32 CharacterID)
+{
+	// Ensure that PlayerState is not null before subscribing to the event
+	if (!ensureMsgf(PlayerState, TEXT("ASSERT: [%i] %hs:\n'PlayerState' is null!"), __LINE__, __FUNCTION__))
```

Comment:
PlayerState guarantees to be valid here, please replace `ensureMsgf` by `checkf`

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/41/commits/48e666e658b15184a1fd1c62ced7ad85224fc25d